ROADMAP.rst

```diff
@@ -0,0 +1,96 @@
+* For learners working beyond basic programming skills it should be simple to switch to "I know what I'm doing" mode. Furthermore, it should be possible to customise Mu to some extent and/or surface more advanced features through editing a settings file.
+* We should encourage and engage with UX specialists. Computing is not just about programming and it would be wonderful if Mu led people to HCI related aspects of our field.
+
+Code Refactor
+=============
+
+* Mu is designed to be simple to code: our hope is that advanced learners will know enough to be able to understand how it works. We want our users to take ownership of the tool. Despite such simplicity there are aspects of the logic and interface code that are not cleanly separated. These should be disentangled (for example, use of Qt serial in the interface layer rather than PySerial in the logic layer).
+* A clean separation of logic and interface means other UI libraries could be used. For example, BeeWare have expressed an interest in helping to port Mu. It also means others have the freedom to change the look and feel of their version of Mu without having to disentangle the logical features.
+* Furthermore, not all the features implemented in the logic layer need be exposed by Mu. Discussion of MegaMu (a more featureful version of Mu) has taken place and providing a selection of features for UI designers to pick'n'mix depending on the needs of their users would make Mu a compelling solution for editor designers.
+* How the logic and interface layers interact should be clearly documented.
+
+Generic Editor
+==============
+
+* The RPi Foundation have asked that we support other Python contexts. For example, having the ability to create / write PyGameZero, GPIOZero and NetworkZero based projects with the UI automatically updating as the context changes. I have an old branch that provides a model for this in the logic layer that reaches into the interface layer. It's an exploratory spike that I believe could bear fruit.
+* To help in the user's graduation to a "real" editor we should ensure generic editor behaviours work as per the conventions: shortcut keys are the same, editing activities (e.g. selecting and acting upon blocks of code) should work as expected and helpful features (like code completion and call tips) should work in "the usual way".
+
+Multi Device Support
+====================
+
+* Adafruit, Calliope and others have expressed an interest in Mu supporting their devices. It should, and do so automagically. It should be possible to use USB's manufacturer and device ID's to work out what's plugged in and adapt accordingly.
```

Comment:
we did a quick hack and this worked out, ideally we or someone can PR something like this ... using this repo ... https://github.com/mu-editor/mu

changed this:
https://github.com/mu-editor/mu/blob/1e3656c93ab2a59a90eeeaf0497fc5fbf527a54c/mu/logic.py
#: USB product ID.
MICROBIT_PID = 516
#: USB vendor ID.
MICROBIT_VID = 3368

to:
#: USB product ID.
MICROBIT_PID = 0x8015
#: USB vendor ID.
MICROBIT_VID = 0x239a

then installed the requirements in requirements.txt and ran python3 run.py to try it out.

works, screenshot attached.

there is a fork of mu that looks for the serial, we would want to use the USB VID/PID instead, but this fork has some updates that fix some terminal stuff too:

https://github.com/eduvik/mu
https://github.com/eduvik/mu/branches

![works](https://cloud.githubusercontent.com/assets/1685947/19831812/78bbb48c-9de1-11e6-83a5-d62e6000d357.png)